File: Python/fib.py
def fibRecursive(n):
    """[summary]
    Computes the n-th fibonacci number recursive.
    Problem: This implementation is very slow.
    approximate O(2^n)

    Arguments:
        n {[int]} -- [description]
    
    Returns:
        [int] -- [description]
    """

    # precondition
    assert n >= 0, 'n must be a positive integer'

    if n <= 1:
        return n
    else:
        return fibRecursive(n-1) + fibRecursive(n-2)

# print (fibRecursive(35))

def fibList(n):
    """[summary]
    This algorithm computes the n-th fibbonacci number
    very quick. approximate O(n)
    The algorithm use dynamic programming.
    
    Arguments:
        n {[int]} -- [description]
    
    Returns:
        [int] -- [description]
    """

    # precondition
    assert n >= 0, 'n must be a positive integer'

    listResults = [0, 1]
    for i in range(2, n+1):
        listResults.append(listResults[i-1] + listResults[i-2])
    return listResults[n]

# print(fibList(100)) # => 354224848179261915075

File: C/fib.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

long long fib_rec(int);
long long fib_list(int); /* dynamic programming */

int 
main(void)
{
    printf("fib_rec(35)=%lld\n",fib_rec(35)); /* => 9227465 */
    printf("fib_list(100)=%lld\n",fib_list(100)); /* => 3736710778780434371 */
    return (0);
}

long long 
fib_rec(int n)
{
    if (n <= 1) {
        return (n);
    } else {
        return (fib_rec(n-1) + fib_rec(n-2));
    }
}

long long
fib_list(int n)
{
    long long list[n+1];
    memset(list, 0, n+1);
    list[0] = 0;
    list[1] = 1;
    int i;

    for (i = 2; i <= n; i++) {
        list[i] = list[i-1] + list[i-2];
    }

    return list[n];
}